CAN Bus Testing

Jac2125 opened the pull request:
# Purpose
Testing for CAN bus

# New Changes
Added CAN testing function 

# Testing
All test passed

dgobalak committed:
Directory revamp (#149)

# Purpose
Change to the directory structure outlined here:
https://www.notion.so/uworbital/FW-Next-Comp-Planning-f04acfffbc6c49fc86d8486fd48e70ab?pvs=4#026f8eb4d991459a8f8ff73fb6876d21
**Note: The Notion page might be out-of-date**

**Goal:** 
- organize the repo so we can put all projects (gs, OBC, etc) in one
repo.
- update the build system to handle tests, OBC fw, and gs code. Our
current build system doesn't use CMake to build the gs.
- Decouple the obc_gs_interface from the OBC fw
- Split things up into more logical sections. I wanted to make sure we
didn't have a situation like last comp where `cdh/` had different things
(command manager, telemetry manager, timekeeper, supervisor) and we
dumped a bunch of stuff into `common/` (some of which can be used by the
ground station like packing utils and others are tied to the OBC fw like
obc_time)

# New Changes
- Changed directory structure; I tried to minimize the number of folders
at the top-level of the directory.
- Switched from include guards to `#pragma once` for a few reasons:
  1. It makes refactoring simpler
2. It's less prone to bugs since you don't have to choose a unique
include guard (that can sometimes be a lot of characters)
  3. It's less code to write
- Moved interface-related comms files to `obc_gs_interface` and added
the `obc_gs_` prefix to each filename
- Added project-specific error codes. `obc_error_code_t` for OBC
firmware, `obc_gs_error_code_t` for interface code, and
`gs_error_code_t` for ground station code. Because of this, I've removed
the use of `RETURN_IF_ERROR_CODE` from the interface for now. In a
future PR, we'll add a logging module that'll work across the obc, gs,
and interface.
- Created standalone obc_gs_interface lib that can be linked into any
executable (obc, gs, or test)
- Removed any dependencies on the main firmware from the
obc_gs_interface module (this allows us to compile it as a static lib)

Other stuff:
- Added telemetry unpack functions
- Added some unit tests (for telemetry pack and obc_time_utils)
- Moved ISR callbacks to `drivers/` to keep the HAL independent from the
rest of our code. This should, in theory, allow us to compile the HAL as
a static lib. I also added back the rtc alarm callback which was deleted
accidentally in a past PR.

# Testing
- CI
- Tested fw on launchpad

# Outstanding Changes
- The CC1120 driver shouldn't be dependent on code from `services/`
- Make a static lib for the hal and then fix the builds for all the test
apps. It'd be nice if we could just link the HAL static lib when
building each test app.
- Create a logging module (with RETURN_IF_ERROR_CODE macros) that'll
work with the different error codes we have
- Write CMake and unit test documentation
- Combine obc_state into supervisor and rethink what telemetry we send
down.

dgobalak committed:
Removed useless extern (#161)

# Purpose
Removed a useless `extern` variable that was in a header file.

# Testing
- CI
- Tested demo on board (with mutex take/release commented out in
`sciSendBytes`)

Jac2125 committed:
CAN bus is tested using canTransmit function

Jac2125 committed:
CAN bus is tested using canTransmit function

Navtajh04 committed:
Remove I2C Critical Sections (#162)

# Purpose
Explain the purpose of the PR here, including references to any existing
Notion tasks.
- Remove unneeded critical sections in obc i2c functions
# New Changes
- Explain new changes

# Testing
- Explain tests that you ran to verify code functionality.

# Outstanding Changes
- If there are non-critical changes (i.e. additional features) that can
be made to this feature in the future, indicate them here.

dgobalak committed:
Removed unnecessary paths from pre-commit exclude list (#164)

# Purpose
pre-commit will only run on files that can be added by Git. So there's
no need to add the build paths in the pre-commit config file since those
paths are already in `.gitignore`.

# Testing
- Ran `pre-commit run --all-files` locally to make sure pre-commit
doesn't run on the build files.

Navtajh04 commented on util/bringup/interface_tests/source/test_can.c:
```diff
@@ -1,4 +1,14 @@
+  sciPrintf("Testing CAN...\r\n");
+  canInit();
+  const uint8 txData = 0xFFU;
+  if (canTransmit(canREG1, 10, &txData) == 0U) {
```
Change the `10` to a macro with a comment explaining it

Jac2125 replied:
I changed this to a macro

Jac2125 committed:
Macro for CAN bus mailbox number defined

Jac2125 committed:
Macro for CAN bus mailbox number defined

manumanuk commented on util/bringup/interface_tests/source/test_can.c:
```diff
@@ -1,4 +1,17 @@
+  canInit();
+  const uint8 txData = 0xFFU;
+  if (canTransmit(canREG1, boxNum, &txData) == 0U) {
+    sciPrintf("Tramsmission Failed.\r\n");
+  }
+
+  sciPrintf("Transmission Succeeded.\r\n");
```
Currently this prints both failed and succeeded on a failure

manumanuk reviewed (changes requested):
Might want to reopen this PR and make a new branch because of the changes to the directory structure

Jac2125 committed:
Had to stop after printing out
message when test fails

Navtajh04 committed:
[BUGFIX] CC1120 TX Semaphores Not Correctly Initialized (#169)

# Purpose
Explain the purpose of the PR here, including references to any existing
Notion tasks.
- Both of the TX semaphores for cc1120 need to be initialized with a
value of 1 otherwise we will never be able to transmit
# New Changes
- Explain new changes

# Testing
- Explain tests that you ran to verify code functionality.
- CI
# Outstanding Changes
- If there are non-critical changes (i.e. additional features) that can
be made to this feature in the future, indicate them here.

Jac2125 committed:
More detailed description and name for Message Box macro added

Jac2125 committed:
Merge branch 'main' into jaemin/CANBringUp

manumanuk commented:
Is this PR still active? Seems like there are basically no file changes compared to main

Jac2125 commented:
I'll close this pr and make a new branch since it looks bit messy right now
